sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -450,14 +450,20 @@ object TypeCoercion {
       case Abs(e @ StringType()) => Abs(Cast(e, DoubleType))
       case Sum(e @ StringType()) => Sum(Cast(e, DoubleType))
       case Average(e @ StringType()) => Average(Cast(e, DoubleType))
-      case StddevPop(e @ StringType()) => StddevPop(Cast(e, DoubleType))
-      case StddevSamp(e @ StringType()) => StddevSamp(Cast(e, DoubleType))
+      case StddevPop(e @ StringType(), nullOnDivideByZero) =>
+        StddevPop(Cast(e, DoubleType), nullOnDivideByZero)
```

Comment:
can we use
```
case s @ StddevPop(e @ StringType(), _) =>
  s.withNewChildren(Seq(Cast(e, DoubleType)))
```

It can retain the `TreeNodeTag` if there are any.